packages/ui/src/components/NavigationMenu/SNavigationMenuItem.vue

```diff
@@ -0,0 +1,63 @@
+<script lang="ts">
+export default { name: 'SNavigationMenuItem' }
+</script>
+
+<script setup lang="ts">
+import { computed, onUnmounted } from 'vue'
+import { useMenuApi, useSubmenuApi } from '@/components/NavigationMenu/api'
+import SNavigationMenuItemBody from '@/components/NavigationMenu/SNavigationMenutemBody.vue'
+
+const props = withDefaults(
+  defineProps<{
+    index: string
+  }>(),
+  {},
+)
+
+const menuApi = useMenuApi()
+
+const isSelected = computed(() => menuApi.active.value === props.index)
+const isMenuCollapsed = computed(() => menuApi.collapsed.value)
+
+function handleClick() {
+  if (menuApi) {
+    menuApi.select(props.index)
+  }
+}
+
+const isInSubmenu = ref(false)
+const submenuApi = useSubmenuApi()
+
+if (submenuApi) {
+  const index = computed(() => props.index)
+  isInSubmenu.value = true
```

Comment:
And again, `isInSubmenu` don't need to be reactive. It's value set during `setup()` and never changes.